test/unit/sources/alternates/twitter_test.rb

```diff
@@ -0,0 +1,143 @@
+require 'test_helper'
+
+module Sources
+  class TwitterTest < ActiveSupport::TestCase
+    context "A mobile twitter link" do
+      setup do
+        @site = Sources::Alternates.find("https://mobile.twitter.com/FalseKnees/status/1555698764622737408")
+      end
+
+      should "be handled by the twitter strategy" do
+        assert(@site.is_a?(Sources::Alternates::Twitter))
+      end
+
+      should "be converted to non-mobile site" do
+        assert_equal(
+          "https://twitter.com/FalseKnees/status/1555698764622737408",
+          @site.original_url
+        )
+      end
+    end
+
+    context "A twitfix link" do
+      setup do
+        @site = Sources::Alternates.find("https://fxtwitter.com/FalseKnees/status/1555698764622737408")
+      end
+
+      should "be handled by the twitter strategy" do
+        assert(@site.is_a?(Sources::Alternates::Twitter))
+      end
+
+      should "be converted to a twitter link" do
+        assert_equal(
+          "https://twitter.com/FalseKnees/status/1555698764622737408",
+          @site.original_url
+        )
+      end
+    end
+
+    context "A twitter link with tracking" do
+      setup do
+        @site = Sources::Alternates.find("https://twitter.com/Idolomantises/status/1554175127855673344?s=20&t=dow0UJIEEOousVoifzpLdg")
+      end
+
+      should "be handled by the twitter strategy" do
+        assert(@site.is_a?(Sources::Alternates::Twitter))
+      end
+
+      should "have the tracking data removed" do
+        assert_equal(
+          "https://twitter.com/Idolomantises/status/1554175127855673344",
+          @site.original_url
+        )
+      end
+    end
+
+    context "A twitter profile link with tracking" do
+      setup do
+        @site = Sources::Alternates.find("https://twitter.com/Idolomantises?s=09")
+      end
+
+      should "have the tracking data removed" do
+        assert_equal(
+          "https://twitter.com/Idolomantises",
+          @site.original_url
+        )
+      end
+    end
+
+    context "An old twitter direct image link" do
+      setup do
+        @site = Sources::Alternates.find("https://pbs.twimg.com/media/E8v96meVgAkTKDE.jpg:orig")
+      end
+
+      should "be handled by the twitter strategy" do
+        assert(@site.is_a?(Sources::Alternates::Twitter))
+      end
+
+      should "convert to use query parameters" do
```

Comment:
Why would you check against two possible matches? Only one of these should be the correct one. Same below.

Reply:
Hmmm, I wasn't sure whether the query order would be guaranteed? The Addressable::URI docs weren't clear, and I wasn't sure Ruby's standards about that.
Trying it both ways around, and throwing some random query values in, it seems like it's encoded alphabetically? At least in the times I tried. I'll just switch stuff to that then

Reply:
Looks like it's sorted alphabeticaly when passing a hash, and respects the ordering when it is an array, like `[["b", "y"], [""a", "z"]]`, `?b=y&a=z`